client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+export const requestResetOptions = ( { dispatch }, { userData } ) => (
+	wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( options => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		options,
```

Comment:
Interesting idea :)
As I can foresee that the latter would be a more reasonable data structure for our need, since the primary set of info and the secondary one will be pumped into the same component. I have added it in 99ccc653dbcef2c7504702511fd8b62c224ec8b7. How does it look? I feel that a pattern of implementing API endpoint interaction in this data layer is emerging :) 

Among writing this, I am wondering if it would be worthwhile to also implement some kinds of data validation in `fromApi`? To me, it looks like we never have such a nice place to do this without disturbing action or state code. I have experienced a few times that an error caused by unexpected response data format change been carried a long way or somehow got consumed by the `catch` handler in some unrelated `Promise`. It would be great if this kind of errors can be caught at early stage. I have no concrete idea of what should it do when the validation failed, though. A warning? A fatal error? Well, It might be not related to this PR, but really fun to tinker about :D